hedera-node/src/main/java/com/hedera/services/contracts/execution/traceability/HederaTracer.java

```diff
@@ -74,35 +75,43 @@ public void traceExecution(MessageFrame frame, ExecuteOperation executeOperation
             if (frameState != State.CODE_EXECUTING) {
                 if (frameState == State.CODE_SUSPENDED) {
                     final var nextFrame = frame.getMessageFrameStack().peek();
-                    trackActionFor(nextFrame, nextFrame.getMessageFrameStack().size() - 1);
+                    trackActionFor(nextFrame, nextFrame.getMessageFrameStack().size() - 1, frame);
                 } else {
                     finalizeActionFor(currentActionsStack.pop(), frame, frameState);
                 }
             }
         }
     }
 
-    private void trackActionFor(final MessageFrame frame, final int callDepth) {
+    private void trackActionFor(
+            final MessageFrame nextFrame, final int callDepth, final MessageFrame parentFrame) {
```

Comment:
I'd prefer to split this into two methods, `trackActionFor(@NotNull final MessageFrame initialFrame)` and `trackActionFor(@NotNull final MessageFrame nextFrame, final int callDepth, @NotNull final MessageFrame parentFrame)`.

Also, since `callDepth = nextFrame.getMessageFrameStack().size() - 1`, does it make sense to pass it as separate parameter?

Reply:
> Also, since callDepth must equal nextFrame.getMessageFrameStack().size() - 1 for the call to be correct, does it make sense to pass it as separate parameter?

Not really 😆 

Tried to clean it up a bit now